frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -302,6 +303,58 @@ FROM scratch AS a
 	checkLinterWarnings(t, sb, &lintTestParams{Dockerfile: dockerfile})
 }
 
+func testJSONArgsRequired(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+FROM scratch
+CMD mycommand
+
+FROM scratch
+ENTRYPOINT mycommand
+`)
+	checkLinterWarnings(t, sb, &lintTestParams{
+		Dockerfile: dockerfile,
+		Warnings: []expectedLintWarning{
+			{
+				RuleName:    "JSONArgsRequired",
+				Description: "JSON arguments required unless a custom shell is specified",
```

Comment:
"required" may be strong word here as build would work fine without it. "preferred/recommended" maybe. Not sure if we need to mention "custom shell" as well as unlikely that the user would know anything about custom shells (and Dockerfile does not contain any `SHELL`). The warning is already not shown if there is a custom shell.